#!/usr/bin/env python
PACKAGE = "apriltags_tas"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

tag_family_enum = gen.enum([gen.const("16h5", int_t, 0, ""),
                            gen.const("25h7", int_t, 1, ""),
                            gen.const("25h9", int_t, 2, ""),
                            gen.const("36h9", int_t, 3, ""),
                            gen.const("36h11", int_t, 4, "")],
                           "")
gen.add("tag_family", int_t, 1, "Tag family", 4, 0, 4, edit_method=tag_family_enum)

gen.add("only_known_tags", bool_t, 0, "Return only tags listed in config/single_tags.txt", False)
gen.add("refine_corners", bool_t, 0, "Refine tag corners by direct edge optimization", True)
gen.add("filter_cross_corners", bool_t, 0, "Filter tags without clean cross corners", True)
gen.add("publish_tf", bool_t, 0, "Publish the tag poses to tf", True)
gen.add("draw_image", bool_t, 0, "Publish image with detected tags", True)

exit(gen.generate(PACKAGE, "apriltags_tas", "AprilTagDetector"))
